bin/solve_stats.py

```diff
@@ -0,0 +1,141 @@
+            submissions[j['submission_id']] = {**j, **submissions[j['submission_id']]}
+
+    bar.finalize()
+
+    ac_teams = {p: set() for p in problems}
+    stats = {p: [{j: 0 for j in judgement_types} for _ in range(bins)] for p in problems}
+    stats_sum = {p: {j: 0 for j in judgement_types} for p in problems}
+    language_stats = {l: {j: 0 for j in judgement_types} for l in languages}
+
+    for i, s in submissions.items():
+        if s['team_id'] not in teams:
+            continue
+        minute = time_string_to_minutes(s['contest_time'])
+        if 0 <= minute < contest_duration:
+            jt = (
+                ''
+                if not post_freeze and minute >= contest_duration - freeze_duration
+                else s['judgement_type_id']
+            )
+            if jt is None:
+                continue
+            if jt == 'AC':
+                ac_teams[s['problem_id']].add(s['team_id'])
+            stats[s['problem_id']][int(minute / scale)][jt] += 1 if jt in ['AC', ''] else -1
+            stats_sum[s['problem_id']][jt] += 1
+            language_stats[s['language_id']][jt] += 1
+
+    problem_stats = ''
+
+    bar = ProgressBar('Plotting', items=[*stats.keys(), 'Language Stats'])
+
+    makedirs(f'solve_stats/activity', exist_ok=True)
+    for p, minutes in stats.items():
+        bar.start(p)
+        label = problems[p]['label']
+        fig, ax = plt.subplots(figsize=(12, 2))
+        # Ugly accumulator. Matplotlib doesn't support negative stacked bars properly: https://stackoverflow.com/a/38900035
+        neg_acc = [0 for m in minutes]
+        # Reverse order, so that the order at the bottom is WA-TLE-RTE
+        for jt in sorted(judgement_types, reverse=True):
+            if jt == 'CE':
+                continue
+            is_neg = any(m[jt] < 0 for m in minutes)
+            ax.bar(
+                range(bins),
+                [m[jt] for m in minutes],
+                1,
+                color=judgement_colors.get(jt) or judgement_colors['RTE'],
+                bottom=neg_acc if is_neg else None,
+            )
+            if is_neg:
+                neg_acc = [a + b for a, b in zip(neg_acc, (m[jt] for m in minutes))]
+        plt.axhline(y=0, linewidth=1, color='gray')
+        ax.autoscale(enable=True, axis='both', tight=True)
+        fig.subplots_adjust(left=0, right=1, bottom=0, top=1)
+        plt.axis('off')
+        plt.tight_layout(pad=0)
+        plt.savefig(f'solve_stats/activity/{label}.pdf', bbox_inches='tight', transparent=True)
+
+        problem_stats += (
+            r'\providecommand{\solvestats'
+            + label
+            + r'}{\printsolvestats{'
+            + '}{'.join(
+                str(x) for x in [sum(stats_sum[p].values()), len(ac_teams[p]), stats_sum[p]['']]
+            )
+            + '}}\n'
+        )
+        bar.done()
+
+    open(f'solve_stats/problem_stats.tex', 'w').write(problem_stats)
```

Comment:
can you do this with `Path(<path>).write_text(..)` instead

Reply:
Costs an extra import, but sure :stuck_out_tongue: